File: src/dataset/javascript/triangle_exercism-ThomasZumsteg/claude/ClaudeSonnet4_triangle_v3.js
var Triangle = function(a, b, c) {
    if (a < 0 || b < 0 || c < 0) 
        throw "Only positive length sides are allowed";
    if (a === 0 || b === 0 || c === 0)
        throw "Triangle cannot have zero area";
    
    var sides = [a, b, c];
    sides.sort(function(x, y) { return x - y; });
    
    if (sides[0] + sides[1] <= sides[2])
        throw "Not a valid triangle";
    
    this.triangle = sides;
    
    if (sides[0] === sides[1] && sides[1] === sides[2])
        this._kind = "equilateral";
    else if (sides[0] === sides[1] || sides[1] === sides[2])
        this._kind = "isosceles";
    else
        this._kind = "scalene";
};

Triangle.prototype.kind = function() {
    return this._kind;
};

module.exports = Triangle;

File: src/dataset/python/triangle_exercism-ThomasZumsteg/claude/ClaudeSonnet4_triangle_v3.py
"""Validate and specify triangle types"""

class TriangleError(Exception):
    """Error if it cannot be a triangle"""
    pass

class Triangle:
    """Validate and specify triangle types"""
    __slots__ = ('a', 'b', 'c')
    
    def __init__(self, a, b, c):
        """Create and validate triangle"""
        sides = sorted((a, b, c))
        self.a, self.b, self.c = sides
        if sides[0] + sides[1] <= sides[2]:
            raise TriangleError("Not a Triangle")

    def kind(self):
        """Specify triangle type"""
        if self.a == self.c:
            return "equilateral"
        elif self.a == self.b or self.b == self.c:
            return "isosceles"
        else:
            return "scalene"